src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -6269,16 +6269,68 @@ def remove_vnet_gateway_aad(cmd, resource_group_name, gateway_name, no_wait=Fals
+def create_virtual_router(cmd,
+                          resource_group_name,
+                          virtual_router_name,
+                          hosted_gateway=None,
+                          hosted_subnet=None,
+                          location=None,
+                          tags=None):
+    vrouter_client = network_client_factory(cmd.cli_ctx).virtual_routers
+    vhub_client = network_client_factory(cmd.cli_ctx).virtual_hubs
+
+    from azure.mgmt.network.models import ErrorException
+    try:
+        vrouter_client.get(resource_group_name, virtual_router_name)
+    except ErrorException:
+        pass
+
+    virtual_hub_name = virtual_router_name
+    try:
+        vhub_client.get(resource_group_name, virtual_hub_name)
+        raise CLIError('The VirtualRouter "{}" under resource group "{}" exists'.format(virtual_hub_name,
+                                                                                        resource_group_name))
+    except CloudError:
+        pass
+
+    SubResource = cmd.get_models('SubResource')
+
+    # for old VirtualRouter
+    if hosted_gateway is not None:
+        VirtualRouter = cmd.get_models('VirtualRouter')
+        virtual_router = VirtualRouter(virtual_router_asn=None,
+                                       virtual_router_ips=[],
+                                       hosted_subnet=None,
+                                       hosted_gateway=SubResource(id=hosted_gateway),
+                                       location=location,
+                                       tags=tags)
+        return vrouter_client.create_or_update(resource_group_name, virtual_router_name, virtual_router)
+
+    # for VirtualHub
+    VirtualHub, HubIpConfiguration = cmd.get_models('VirtualHub', 'HubIpConfiguration')
+
+    hub = VirtualHub(tags=tags, location=location, virtual_wan=None, sku='Standard')
+    ip_config = HubIpConfiguration(subnet=SubResource(id=hosted_subnet))
+
+    from azure.cli.core.commands import LongRunningOperation
+
+    vhub_poller = vhub_client.create_or_update(resource_group_name, virtual_hub_name, hub)
+    LongRunningOperation(cmd.cli_ctx)(vhub_poller)
+
+    vhub_ip_config_client = network_client_factory(cmd.cli_ctx).virtual_hub_ip_configuration
+    try:
+        vhub_ip_poller = vhub_ip_config_client.create_or_update(resource_group_name,
+                                                                virtual_hub_name,
+                                                                'Default',
+                                                                ip_config)
+        LongRunningOperation(cmd.cli_ctx)(vhub_ip_poller)
+    except Exception as ex:
+        logger.error(ex)
+        vhub_ip_config_client.delete(resource_group_name, virtual_hub_name, 'Default')
+        vhub_client.delete(resource_group_name, virtual_hub_name)
+        raise ex
+
+    return vhub_client.get(resource_group_name, virtual_hub_name)
 
 
 def update_virtual_router(cmd, instance, tags=None):
```

Comment:
Should this changed as well? well. We can pass this one this time since it's not a breaking change for existing users.

Reply:
Added